### PR TITLE
Added get_daily_prices interface to price source.

### PR DESCRIPTION
**Pull request by blais**
Implemented for Yahoo.

I think eventually we'll want a richer interface to the price source, with a frequency spec.

### CODE REVIEW

**Review by xuhcc — COMMENTED** [review]
pylint report:
```
************* Module beanprice.sources.yahoo
beanprice/sources/yahoo.py:92: [C0103(invalid-name), get_price_series] Variable name "tz" doesn't conform to '(_?[a-z_][a-z0-9_]{2,30}|__|mu|no)$' pattern
beanprice/sources/yahoo.py:123: [C0103(invalid-name), Source.get_latest_price] Variable name "tz" doesn't conform to '(_?[a-z_][a-z0-9_]{2,30}|__|mu|no)$' pattern
beanprice/sources/yahoo.py:157: [E0602(undefined-variable), Source.get_daily_prices] Undefined variable 'time'
beanprice/sources/yahoo.py:157: [E0602(undefined-variable), Source.get_daily_prices] Undefined variable 'time'

------------------------------------------------------------------
Your code has been rated at 9.91/10 (previous run: 9.96/10, -0.05)
```

There are also several type errors (I'm using mypy to check):
```
$ mypy  --ignore-missing-imports beanprice
beanprice/sources/yahoo.py:63: error: Incompatible return value type (got "None", expected "str")
beanprice/sources/yahoo.py:64: error: Argument 2 to "get" of "Mapping" has incompatible type "None"; expected "str"
beanprice/sources/yahoo.py:88: error: Argument "params" to "get" has incompatible type "Dict[str, object]"; expected <union: 9 items>
beanprice/sources/yahoo.py:101: error: Incompatible return value type (got "Tuple[List[Tuple[datetime, Any]], Any]", expected "List[Tuple[datetime, Decimal]]")
beanprice/sources/yahoo.py:142: error: 'builtins.object*' object is not iterable
beanprice/sources/yahoo.py:142: error: Value of type variable "_LT" of "sorted" cannot be "object"
beanprice/sources/yahoo.py:143: error: Cannot determine type of 'data_dt'
beanprice/sources/yahoo.py:145: error: Cannot determine type of 'data_dt'
beanprice/sources/yahoo.py:145: error: Cannot determine type of 'price'
beanprice/sources/yahoo.py:149: error: Cannot determine type of 'price'
beanprice/sources/yahoo.py:149: error: Cannot determine type of 'data_dt'
beanprice/sources/yahoo.py:149: error: Argument 3 to "SourcePrice" has incompatible type "Tuple[datetime, Decimal]"; expected "Optional[str]"
beanprice/sources/yahoo.py:157: error: Name 'time' is not defined
beanprice/sources/yahoo.py:159: error: 'builtins.object' object is not iterable
beanprice/sources/yahoo.py:159: error: Cannot determine type of 'price'
beanprice/sources/yahoo.py:159: error: Cannot determine type of 'time'
beanprice/sources/yahoo.py:159: error: Argument 3 to "SourcePrice" has incompatible type "Tuple[datetime, Decimal]"; expected "Optional[str]"
```

File: beanprice/sources/yahoo.py [hunk 1]
```diff
@@ -70,10 +71,40 @@ def parse_currency(result: Dict[str, Any]) -> str:
 }
 
 
+def get_price_series(ticker: str,
+                     time_begin: datetime,
+                     time_end: datetime) -> List[Tuple[datetime, Decimal]]:
```
Comment (xuhcc) [inline beanprice/sources/yahoo.py:76]:
```suggestion
                     time_end: datetime) -> Tuple[List[Tuple[datetime, Decimal]], str]:
```

**Review by xuhcc — COMMENTED** [review]
I can confirm that yahoo source is working.

File: beanprice/sources/yahoo.py (hunk 1 above)
Comment (blais) [reply beanprice/sources/yahoo.py:76]: Good catch. Done. Will upload and merge.
I'll eventually integrate pytype() in the Bazel build, so it'll run automatically.

**Comment by blais** [comment]
BTW, beyond unit tests, we should start building a test suite to be run manually every now and then during the week, which actually hits the sources. That could be useful to find upstream changes.

**Pull request merged**